### PR TITLE
cpp-ethereum:develop integration

### PR DESCRIPTION
**Pull request by winsvega**
in development

a script to run eth in hive.

### CODE REVIEW

**Comment by karalabe** [comment]
Uuuu, awesome! :D Tell me when it's ready for merge :)

File: clients/cpp-ethereum:develop/eth.sh [hunk 1]
```diff
@@ -0,0 +1,87 @@
+#!/bin/bash
+
+# Startup script to initialize and boot a go-ethereum instance.
```
Comment (chriseth) [inline clients/cpp-ethereum:develop/eth.sh]: rename

File: clients/cpp-ethereum:develop/eth.sh [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+#
+# This script assumes the following files:
+#  - `geth` binary is located in the filesystem root
```
Comment (chriseth) [inline clients/cpp-ethereum:develop/eth.sh]: rename

File: clients/cpp-ethereum:develop/eth.sh [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+#  - HIVE_MINER          address to credit with mining rewards (single thread)
+#  - HIVE_MINER_EXTRA    extra-data field to set for newly minted blocks
+
+# Immediately abort the script on any error encountered
+set -e
+ETH=cpp-ethereum/build/eth/eth
+
+# It doesn't make sense to dial out, use only a pre-set bootnode
+if [ "$HIVE_BOOTNODE" != "" ]; then
+	FLAGS="$FLAGS --peerset $HIVE_BOOTNODE"
```
Comment (chriseth) [inline clients/cpp-ethereum:develop/eth.sh]: Do you also need `--pin`?

File: clients/cpp-ethereum:develop/eth.sh [hunk 4]
```diff
@@ -0,0 +1,87 @@
...
+fi
+
+# If the client is to be run in testnet mode, flag it as such
+if [ "$HIVE_TESTNET" == "1" ]; then
+	FLAGS="$FLAGS --morden"
+fi
+
+# Handle any client mode or operation requests
+if [ "$HIVE_NODETYPE" == "light" ]; then
+	FLAGS="$FLAGS --mode peer"
```
Comment (chriseth) [inline clients/cpp-ethereum:develop/eth.sh]: Is that really the correct setting, or is that about light client mode?

**Comment by holiman** [comment]
This file does not import accounts from the `genesis` supplied from hive. At least as far as I can see (?)

Check https://github.com/holiman/hive/blob/hive_blocktests/clients/parity:master/parity.sh#L43 to see how it's done in parity, or https://github.com/holiman/hive/blob/hive_blocktests/clients/go-ethereum:master/geth.sh#L72 how geth does it.

**Comment by winsvega** [comment]
This file should describe cpp-ethereum as a client. 
Account import is done in the hive test case.

**Comment by karalabe** [comment]
@holiman I guess this PR can be closed since your PR did a full CPP integration? Please reopen if not.

**Pull request closed without merging**